packages/marked-mathjax/src/index.ts

```diff
@@ -0,0 +1,76 @@
+import type { Marked } from 'marked';
+
+const startMath = /(\$|\\\(|\\\[)/;
+
+/**
+ * Adds an extension to marked to handle math in a manner compatible with our
+ * use of Mathjax in PrairieLearn.
+ *
+ * @param marked - The marked instance to extend.
+ * @param MathJax - The MathJax instance to use for rendering math. In
+ * client-side code, this is `window.MathJax`. In server-side code, this
+ * is the result of `require('mathjax-full').init()`.
+ */
+export function addMathjaxExtension(marked: Marked, MathJax: any) {
+  const mathjaxInput = MathJax.startup.getInputJax() ?? [];
```

Comment:
Is there any potential for a race condition here where this is undefined? This isn't part of the `MathJax.startup` object that we ourselves attach to the window, so I this is always relying on MathJax having loaded:

https://github.com/PrairieLearn/PrairieLearn/blob/537588c0d86d9c5e5a53bd303f26cb7e448cd16d/apps/prairielearn/assets/scripts/lib/mathjax.ts#L46-L67

Reply:
To ensure we don't end up with a race condition I have added a call to `MathJax.typesetPromise()` before every call to `addMathjaxExtension` in ee943d3cf684c2c070c1f0e3a280749f7c484a32.

I considered making this call inside `addMathjaxExtension` itself, but this could be a problem because, AFAIK, mathjax replaces `window.MathJax` with a new object instead of updating it, meaning the argument to `addMathjaxExtension` would no longer match `window.Mathjax` if the promise was unresolved on call.

Reply:
`MathJax.startup.promise` is probably a better thing to await. That'll resolve once things are up and running without actually re-typesetting the entire document.

Agreed that we shouldn't do that in `addMathjaxExtension`. I think that would in theory complicate server-side things.

Reply:
Good point, I forgot that there were two different promises in play. Changed in 4d5ebe79d5c516b50025de71ee40b6829230c853.